jetty-documentation/src/main/asciidoc/administration/sessions/session-hierarchy.adoc

```diff
@@ -0,0 +1,51 @@
+//  ========================================================================
+//  Copyright (c) 1995-2016 Mort Bay Consulting Pty. Ltd.
+//  ========================================================================
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+
+[[jetty-sessions-architecture]]
+=== Session Architecture
+
+==== Changes in Session Architecture
+The architecture of Session Management Jetty changed significantly in Jetty 9.4.
+These changes have resulted in Sessions not only being easier to configure but making them much more pluggable for various technologies.
+
+In previous versions of Jetty, users were required to have a separate `SessionIdManager` for each kind of session manager implemented (JDBC, MongoDB..etc.); now there is a singular `SessionIdManager` for the entire Jetty instance.
+Likewise, prior to Jetty 9.4 there were several different instances of the `SessionManager` class.
+Instead of a single `SessionManager` though, it has been done away with entirely, with most of it's functionality moved to the `SesssionHandler` class.
+Additionally, Jetty 9.4 introduced the concepts of a `SessionCache` and an associated `SessionDataStore` (both explained below).
+
+Finally, Session scavenging has been re-worked.
+Where previously each `SessionManager` instance would periodically scan the in-memory (or clustered) sessions for expired sessions, there is now a single generic scavenger thread which instructs the `SessionHandler` to clean up expired sessions.
+
+
+==== Session Architecture Hierarchy
+
+Each Jetty instance has a singular `SessionIdManager` to handle all session requests, regardless of clustering technology.
+For each context on the server there is one (1) `SessionCache` which acts as a cache of Session objects for the given context.
+The purpose of the `SessionCache` is to ensure that simultaneous requests always operate on the same Session object.
```

Comment:
The benefit of the 'SessionCache' is to ensure that simultaneous requests always operate on the same Session object.